### ISSUE TITLE
Mi Band 2: Notified twice when recieving chat message

### ISSUE THREAD

**Issue by mueller-ma** [Freeyourgadget/Gadgetbridge#754]
#### Before opening an issue please confirm the following:
- [x] I have read the [wiki](https://github.com/Freeyourgadget/Gadgetbridge/wiki), and I didn't find a solution to my problem / an answer to my question.
- [x] I have searched the [issues](https://github.com/Freeyourgadget/Gadgetbridge/issues), and I didn't find a solution to my problem / an answer to my question.
- [x] If you upload an image or other content, please make sure you have read and understood the [github policies and terms of services](https://help.github.com/articles/github-terms-of-service/#1-responsibility-for-user-generated-content)

#### Your issue is:
When I send via Debug (also applies to real signal notifications) a Signal/FB/FB_Messenger/Twitter/Riot/Telegram/WA message to my mi band I first get a chat icon "two bubbles" and it vibrates twice and after that I get the animated sms icon (one square bubble) it vibrates again.

#### Your wearable device is:
Mi Band 2
FW: 1.0.1.47
HW: v0.1.3.2
#### Your android version is:
Android 7.1.2
#### Your Gadgetbridge version is:
0.19.4

**Comment by mueller-ma** [comment]
This effect does not appear when recieving threema mesages. There i just see the animated one bubble icon and two short vibrations

**Comment by mueller-ma** [comment]
When I disable textual messages everything seems to work fine. I think I must have a deeper look into textual messages

**Comment by ashimokawa** [comment]
@mueller-ma 
I think the code - for some reasons - only uses textual notifications on specific known notification sources (like from Signal), in that case somehow a vibration and the textual notification is sent, the vibration seems to cause the exta two bubbles empty notificaion. Threema is not known so it only uses the bubble notificaion without text.
You can reproduce the behaviour from the Gadgetbridge Debug Activity, so I removed the android integration tag (I am 99% sure it has nothing to do with that)

@cpfeiffer 
Is there any reason for only using textual notificaions for known sources?
And: Why is there an extra notificaions for textual ones? Is it because there was a firmware where we had to do that so it vibrates at all?

**Comment by mueller-ma** [comment]
I have read that I need to install some fonts to use textual notifications. Do you know if they are already on a stock band installed?

**Comment by ashimokawa** [comment]
@mueller-ma 
Sorry, I do not know. I do not own a Mi Band 2.
I tested that on an Amazfit Bip, and I assume you and me have the same bug regarding double notification because FW  1.0.1.47 changed behavior and is now similar to Amazfit Bip.

I guess older firmwares did not vibrate when sending a textual notification so both had been done (vibrate and then sending text), now sending text vibrates and "vibrating only " now also sends the double bubble.

I guess @cpfeiffer can shed some light on the issue

**Comment by cpfeiffer** [comment]
I'm trying to explain the issue, but if I'm missing the point, please tell.

With Mi Band 1*, we had fine grained control over notifications. We could actually tell the device to start vibrating, stop vibrating, blinking in a specific color (first Mi Band 1 only), rhythm, etc. That brought us the "vibration patterns" in the preferences.

At some point, the BLE API of the Mi devices was changed. Newer firmwares, newer devices only support the standard BLE notification APIs. We tried to simulate vibration patterns with these APIs (see those NotificationStrategy* classes), but it really doesn't work very well, since these APIs often also display an icon or text, which then appear and disappear in the rhythm of the vibration pattern.

Then we added support for text notifications on the Mi2, which complicated things further. At least with the newest firmware I had, when I implemented it, the Mi2 showed one specific icon when you created a text notification. I found this problematic, because then you couldn't distinguish Signal, SMS, Mail and other notifications from one another.

So my solution was to first vibrate and display the tool-specific notification icon (honoring the configured vibration pattern) and then display the actual text notification (with the unchangeable "message" icon).

We might revise this, if newer firmware versions behave differently, or if you can come up with a better idea.

We might also drop the notification pattern completely for Mi2, since it really doesn't work very well.

Suggestions welcome.

**Comment by cpfeiffer** [comment]
@mueller-ma yes, you need to install Mili_pro.ft.en to your Mi Band 2. At least my band didn't have it pre-installed. OTOH, when it was manufactured, the text notification feature did not even exist, so maybe newer produced bands already have it.

**Comment by ashimokawa** [comment]
@cpfeiffer
OK got it.
One more question: why did you choose not to support textual notifications for unknown notification sources and navigation type notification?
That's why I don't get any notifications on bip for these now. (I stripped the "vibrate only" command before the textual stuff and the textual stuff is not done for these)

**Comment by mueller-ma** [comment]
I think the best would be to replace several configuration menus and replace it with one: a list with all apps where you can set mute, text notification, icon, vibration and other stuff for each app

**Comment by mueller-ma** [comment]
I flashed it and now it works

**Comment by mueller-ma** [comment]
To be honest I would rather see the animated bubble and after that the one bubble with text, The two bubble icon looks terrible
